.github/scripts/check_base_image_tags/check_base_image_tags.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+"""
+Check that base_image references to pipelines-components images use :main tag.
+
+This ensures components reference the latest images rather than specific SHAs or versions.
+The CI will override these with PR-specific tags during validation.
+"""
+
+import argparse
+import re
+import sys
+from pathlib import Path
+
+
+def find_base_image_references(file_path: Path, image_prefix: str) -> list[tuple[int, str]]:
+    """
+    Find lines containing base_image references to the given image prefix.
+
+    Args:
+        file_path: Path to the Python file to scan.
+        image_prefix: The image prefix to look for (e.g., ghcr.io/kubeflow/pipelines-components).
+
+    Returns:
+        List of tuples (line_number, line_content) for matching lines.
+
+    Raises:
+        OSError: If the file cannot be read.
+        UnicodeDecodeError: If the file contains invalid encoding.
+    """
+    matches = []
+    content = file_path.read_text()
+    for line_num, line in enumerate(content.splitlines(), start=1):
+        if "base_image" in line and image_prefix in line:
+            matches.append((line_num, line))
+    return matches
+
+
+def check_main_tag(line: str, image_prefix: str) -> tuple[bool, str | None]:
+    """
+    Check if a line uses the :main tag for the image.
+
+    Args:
+        line: The line content to check.
+        image_prefix: The image prefix to look for.
+
+    Returns:
+        Tuple of (is_valid, image_reference). is_valid is True if :main tag is used.
+    """
+    escaped_prefix = re.escape(image_prefix)
+    main_tag_pattern = rf"{escaped_prefix}-[^:\"']+:main"
```

Comment:
the pattern here might not validate because of the hypen, especially for simple base images (`python`, `mysql`, etc)